sign/request.go

```diff
@@ -0,0 +1,36 @@
+package sign
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pborman/uuid"
+	"github.com/status-im/status-go/geth/account"
+)
+
+type completeFunc func(*account.SelectedExtKey) (common.Hash, error)
+
+// Meta represents any metadata that could be attached to a signing request.
+// It will be JSON-serialized and used in notifications to the API consumer.
+type Meta interface{}
+
+// Requests is single signing request.
+type Request struct {
+	ID           string
+	Meta         Meta
+	context      context.Context
+	locked       bool
+	completeFunc completeFunc
+	result       chan Result
+}
+
+func newRequest(ctx context.Context, meta Meta, completeFunc completeFunc) *Request {
+	return &Request{
+		ID:           uuid.New(),
+		Meta:         meta,
+		context:      ctx,
+		locked:       false,
+		completeFunc: completeFunc,
+		result:       make(chan Result, 1),
```

Comment:
If a buffered channel is used, then `go func() { request.result <- result }()` does not need to be called in a goroutine. Personally, I prefer unbuffered channels so I would change this line.

Reply:
Because if the business logic is wrong, the same problem will be experienced with buffered channels just a bit later. Buffered channels are cool if there is a typical producer-consumer problem but I don't see it to be the case here.